src/main/java/com/marklogic/kafka/connect/source/RowBatcherSourceTask.java

```diff
@@ -54,63 +48,54 @@ public final void start(Map<String, String> config) {
         logger.info("Starting RowBatcherSourceTask");
         Map<String, Object> parsedConfig = MarkLogicSourceConfig.CONFIG_DEF.parse(config);
         DatabaseClientConfig databaseClientConfig = new DefaultDatabaseClientConfigBuilder().buildDatabaseClientConfig(parsedConfig);
-        this.databaseClient = new DefaultConfiguredDatabaseClientFactory().newDatabaseClient(databaseClientConfig);
+        databaseClient = new DefaultConfiguredDatabaseClientFactory().newDatabaseClient(databaseClientConfig);
         dataMovementManager = databaseClient.newDataMovementManager();
         pollDelayMs = (Long) parsedConfig.get(MarkLogicSourceConfig.WAIT_TIME);
-        this.constraintColumn = (String) parsedConfig.get(MarkLogicSourceConfig.CONSTRAINT_COLUMN_NAME);
-        if (!StringUtils.hasText(constraintColumn)) {
-            constraintColumn = null;
-        }
 
-        queryContextBuilder = AbstractRowBatchBuilder.newQueryContextBuilder(dataMovementManager, parsedConfig);
+        queryHandler = QueryHandler.newQueryHandler(databaseClient, parsedConfig);
```

Comment:
I don't think this should be initialized on `start`, but rather it should be created each time (like RBB) in `poll`. I think the flow should be in `poll`:

1. Call `QueryHandler.newQueryHandler` to obtain a new QH (which will only be a local variable in `poll`)
2. Call `addQueryToRowBatcher`, passing in the "previous max value" which the Task is responsible for obtaining access to (since it will eventually live in ML and be retrieved from there on every `poll` call) . As part of this call, the QH modifies the user's query to include the constraint column and previous max value (both of which may be null, in which case it does nothing).
3. Task runs the job
4. Task then calls a `getMaxConstraintColumnValue` function on QH that uses the modified user query to get the max value, and the Task stores that 

The QH thus lives only for the duration of the `poll()` method and thus should not be a class attribute here. And while the QH is responsible for getting the max value, the Task is responsible for storing it - currently as a Task class attribute, eventually and optionally as a value in ML.